core-bundle/src/Resources/contao/library/Contao/User.php

```diff
@@ -561,6 +561,7 @@ public function serialize()
 			'username' => $this->username,
 			'password' => $this->password,
 			'admin' => $this->admin,
+			'amg' => $this->amg,
```

Comment:
Why is this necessary? It will invalidate the token if the allowed member groups change.

Reply:
If the "amg" property gets not serialized, the property is not present when checking/populating the token, and as a consequence, this will fail/never be the case: https://github.com/contao/contao/pull/989/commits/dfe0d2a984e6c3117bcc75a566bb0e13c96389de#diff-8b7196803029c7e93c87690d75f8ff91R590

Reply:
As discussed in Mumble on December 19th, this is correct, because Symfony does not allow to refresh roles on the fly.